app/policies/article_policy.rb

```diff
@@ -0,0 +1,16 @@
+class ArticlePolicy
```

Comment:
You need to inherit from `ApplicationPolicy`, remove the `def initialize` and the `attr_reader`

Then you can do the following with `update?` and `destroy?`

```ruby
def update?
  user == record.try(:user)
end
```

Reply:
Done. Took my slow ass a while to understand.